### PR TITLE
Symfony 5 upgrade

### PR DESCRIPTION
**Pull request by Shaky212**
A mighty PR where multiple disciplines entailing the Symfony upgrade come together.

Among them:
#562 Upgrade to Symfony 4 (in preparation to this SF5 upgrade)
#571 Move to Github actions for test integrations

At first these tasks where created separate. But to keep things more tight, I chose to merge/rebase them into this branch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,7 @@
 /.vagrant/
```
Comment (MKodde) [inline .gitignore:1]: All traces of Vagrant can be removed in my opinion. But feel free to do this in a new PR!

File: .gitignore [hunk 2]
```diff
@@ -16,8 +15,41 @@ app/js/coverage
...
 /vendor/
-/web/bundles/
 /.idea
 /node_modules
+/.env
+###> symfony 4 ###
+/public/build/
+###< symfony 4 ###
 /web/build
 /web/coverage
```
Comment (MKodde) [inline .gitignore:25]: The /web folder is a SF3 folder right?

File: ansible/roles/spdashboard/tasks/install-branch.yml [hunk 3]
```diff
@@ -40,7 +40,7 @@
 - name: Delete config
   file: path={{spdashboard_data_dir}}/current/app/{{ item }} state=absent
```
Comment (MKodde) [inline ansible/roles/spdashboard/tasks/install-branch.yml:41]: Please discuss this with @quartje, but I think we can let go of the `/ansible` roles in the project. At the very least, our dev roles can be removed as we use the Docker solution exclusively nowadays.

File: public/index.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?php
+//use App\Kernel;
```
Comment (MKodde) [inline public/index.php:2]: The copyright block is missing, and the commented Kernel can be removed.

**Review by MKodde — COMMENTED** [review]
What an effort! Hats off to you @Shaky212 

See some feedback that you might need to do something with, but at the other hand. Might also want to fix in a successive PR. You be the judge.

For now, we should first get a 4.4 release before merging this to `develop`.

File: docker/conf/000-default-dev.conf [hunk 5]
```diff
@@ -1,6 +1,7 @@
 DocumentRoot "/var/www/html/public"
 SetEnvIfNoCase ^Authorization\$ "(.+)" HTTP_AUTHORIZATION=\$1
 SetEnv HTTPS on
+Mutex posixsem
```
[the remark this answers is not in the document]
Comment (Shaky212) [reply docker/conf/000-default-dev.conf:4]: I understood from @quartje that this setting can do no harm, but I will remove it to be sure.

File: .gitignore (hunk 1 above)
Comment (Shaky212) [reply .gitignore:1]: @quartje confirmed that ansible / vagrant can be removed from the premises.

File: .gitignore (hunk 2 above)
Comment (Shaky212) [reply .gitignore:25]: Yes...

File: .env~Stashed changes [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# In all environments, the following files are loaded if they exist,
```
[the remark this answers is not in the document]
Comment (Shaky212) [reply .env~Stashed changes:1]: Yes, will be removed

File: build.xml [hunk 7]
```diff
@@ -120,7 +120,7 @@
         </exec>
     </target>
 
-    <target name="web-tests-no-coverage" depends="create-test-db">
+    <target name="web-tests-no-coverage">
```
[the remark this answers is not in the document]
Comment (Shaky212) [reply build.xml]: build.xml is absolete

File: .gitignore (hunk 2 above)
Comment (Shaky212) [reply .gitignore:25]: Yes, /public from SF4+

File: ansible/roles/spdashboard/tasks/install-branch.yml (hunk 3 above)
Comment (Shaky212) [reply ansible/roles/spdashboard/tasks/install-branch.yml:41]: Confirmed by @quartje remove ansible/vagrant

File: composer.json [hunk 8]
```diff
@@ -53,59 +68,31 @@
...
-    "symfony/polyfill-php70": "*",
-    "symfony/polyfill-php56": "*"
-  },
   "scripts": {
-    "auto-scripts": {
-      "cache:clear": "symfony-cmd",
-      "assets:install %PUBLIC_DIR%": "symfony-cmd"
-    },
+    "symfony-scripts": [
+      "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
```
Comment (MKodde) [inline composer.json:86]: This Symfony script will not work anymore as we removed the parameters handler dependency.

File: composer.json [hunk 9]
```diff
@@ -53,59 +68,31 @@
...
+      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
+      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
+      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile"
```
Comment (MKodde) [inline composer.json:89]: Same for the DistributionBundle scripts. Output of `composer install`

```
Class Incenteev\ParameterHandler\ScriptHandler is not autoloadable, can not call symfony-scripts script
Class Sensio\Bundle\DistributionBundle\Composer\ScriptHandler is not autoloadable, can not call symfony-scripts script
Class Sensio\Bundle\DistributionBundle\Composer\ScriptHandler is not autoloadable, can not call symfony-scripts script
Class Sensio\Bundle\DistributionBundle\Composer\ScriptHandler is not autoloadable, can not call symfony-scripts script
```

File: composer.json [hunk 10]
```diff
@@ -161,6 +148,13 @@
...
       "require": "5.4.*"
-    }
+    },
+    "symfony-app-dir": "app",
+    "symfony-bin-dir": "bin",
+    "symfony-var-dir": "var",
+    "symfony-web-dir": "web",
+    "symfony-tests-dir": "tests",
+    "symfony-assets-install": "relative",
+    "branch-alias": null
```
Comment (MKodde) [inline composer.json:158]: This is SF3 stuff..

**Review by MKodde — COMMENTED** [review]
I only reviewed the newly added commits and found some merge issues in the `composer.json`.

**Review by MKodde — APPROVED** [review]
What a pickle, thanks for persevering on this one!

**Pull request merged**